include/dali/c_api.h

```diff
@@ -42,6 +42,12 @@ typedef enum {
   GPU = 1
 } device_type_t;
 
+typedef enum {
+  DALI_BACKEND_GPU = 0,
+  DALI_BACKEND_CPU = 1,
+  DALI_BACKEND_MIXED = 2
+} dali_backend_t;
```

Comment:
Couple things here. 

In `device_type_t` we assign `CPU=0` and `GPU=1` and here it's reversed. How about making this consistent?
```C++
CPU=0, GPU=1, MIXED=2
```

Secondly, in `device_type_t` we don't use `DALI` prefix for the enum name. Maybe we should keep this consistent also? After all, there's `dali` already in type name, so it's still readable:
```c++
typedef enum {
  CPU = 0,
  GPU = 1,
  MIXED = 2
} dali_backend_t;

dali_backend_t::CPU;
dali_backend_t::GPU;
dali_backend_t::MIXED;
```

Reply:
It's device type that is inconsistent with the rest of the enums. Also, C enums without a prefix is a bad idea, because you can reference them without device_type_t::*, so our namespace is polluted with `CPU` and `GPU`.

Reply:
I can't even name those `CPU` and `GPU`. The names are already taken by the device_type_t

Reply:
> It's device type that is inconsistent with the rest of the enums

In C API I see only `device_type_t` enum so far, so maybe it's good idea to keep them consistent within single API?

Reply:
About the reverse order - it's taken from OpType

Reply:
> I'm wondering, that in C API I see only `device_type_t` enum so far, so maybe it's good idea to keep them consistent within single API?

```c++
typedef enum {
  DALI_NO_TYPE  = -1,
  DALI_UINT8    =  0,
  DALI_UINT16   =  1,
  DALI_UINT32   =  2,
  DALI_UINT64   =  3,
  DALI_INT8     =  4,
  DALI_INT16    =  5,
  DALI_INT32    =  6,
  DALI_INT64    =  7,
  DALI_FLOAT16  =  8,
  DALI_FLOAT    =  9,
  DALI_FLOAT64  =  10,
  DALI_BOOL     =  11,
} dali_data_type_t;
```

Reply:
I changed the order of the GPU and CPU